src/server/api/schema.js

```diff
@@ -1039,40 +1154,167 @@ const rootMutations = {
           interaction_step_id: interactionStepId,
           value
         }).save()
-        const interactionStepResult = await r.knex('interaction_step')
-        // TODO: is this really parent_interaction_id or just interaction_id?
-          .where({ 'parent_interaction_id': interactionStepId,
-                  'answer_option': value })
+        const interactionStepResult = await r
+          .knex('interaction_step')
+          // TODO: is this really parent_interaction_id or just interaction_id?
+          .where({
+            parent_interaction_id: interactionStepId,
+            answer_option: value
+          })
           .whereNot('answer_actions', '')
           .whereNotNull('answer_actions')
 
-        const interactionStepAction = (interactionStepResult.length && interactionStepResult[0].answer_actions)
+        const interactionStepAction =
+          interactionStepResult.length && interactionStepResult[0].answer_actions
         if (interactionStepAction) {
           // run interaction step handler
           try {
             const handler = require(`../action_handlers/${interactionStepAction}.js`)
             handler.processAction(qr, interactionStepResult[0], campaignContactId)
           } catch (err) {
-            console.error('Handler for InteractionStep', interactionStepId,
-                          'Does Not Exist:', interactionStepAction)
+            console.error(
+              'Handler for InteractionStep',
+              interactionStepId,
+              'Does Not Exist:',
+              interactionStepAction
+            )
           }
         }
       }
 
       const contact = loaders.campaignContact.load(campaignContactId)
       return contact
+    },
+    reassignCampaignContacts: async (
+      _,
+      { organizationId, campaignIdsContactIds, newTexterUserId },
```

Comment:
Since this is a pretty powerful api and a somewhat noisy one (all the contact ids) -- it might be good to include a bit of redundancy on the inputs.  e.g. for the use cases where it's always the same oldTexterUserId and/or from a single campaign (or even the campaign id per contact) -- maybe we should take those as inputs to filter more aggressively -- just to make sure a bug doesn't easily sneak in to the frontend and accidentally reassign a bunch of random stuff.

Reply:
@schuyler1d since the api only reassigns CampaignContacts explicitly listed in the campaignIdsContactIds parameter, the reassignment is tightly constrained as it stands, in terms of input.

Ok, I see. Are you suggesting that we filter more aggressively in the sql statements we're using for the reassignment ?

Reply:
👍